### ISSUE TITLE
Gradients are [None,...] in text GAN model

### ISSUE THREAD

**Issue by e8panse** [keras-team/tf-keras#293]
I'm trying to build text GAN using TF2. But gradients_of_generator always becoming [None].  Provided code below uses paddings in between generated_prediction and discriminator input. BUT even if I directly pass generator output to discriminator input (of course changing input_length of discriminator), gradients always [None].

How compute gradient in such text GAN correclty?

Please notice, I simplifed code to give more clear understanding of the problem.

code:

`generator = keras.Sequential(
    [
        layers.Embedding(total_words, 80, input_length=max_sequence_length-1),
        LSTM(100, return_sequences=True),
        LSTM(50),
        tf.keras.layers.Dropout(0.1),
        Dense(total_words/20),
        Dense(total_words, activation='softmax'),
    ],
    name="generator",
)

discriminator = keras.Sequential(
    [
        layers.Embedding(total_words, 80, input_length=max_sequence_length-1),
        layers.Bidirectional(tf.keras.layers.LSTM(64)),
        layers.Dropout(0.2),
        layers.Dense(64, activation='relu'),
        layers.Dense(1, activation='sigmoid'),
    ],
    name="discriminator",
)

    #function: train_step
    #text - tokenized text

    with tf.GradientTape() as gen_tape:
        
        predicted_next_word = generator(text, training=True)    
        pred_to_word = pad_sequences([[np.argmax(predicted_next_word, axis=-1)]], maxlen=max_sequence_length-1, padding='pre')
        fake_output = discriminator(pred_to_word, training=True)
        generator_loss = binary_cross_entropy(tf.ones(fake_output.shape), fake_output) #gives correct loss results
        
    gradients_of_generator = gen_tape.gradient(generator_loss, generator.trainable_variables)`

Gradient are [None,...]
